packages/engine/lib/stateful/src/context/mod.rs

```diff
@@ -1,26 +1,237 @@
-// TODO: DOC: Add module level docs for describing the high level concept of the context, what they
-//   are and why they exist
+//! The global state shared between all [`Agent`]s.
```

Comment:
I don't know how to feel about this (and the other docs with it in like messages). I'm not sure a link to [`Agent`] is helpful here. The `Agent` struct is basically just a JSON serialisable thing, it's not an actual Agent object and isn't used in the runners, etc. It doesn't have any methods associated to agent behavior execution and so on.

Maybe there isn't a good solution to this because we don't have a proper struct to point to, and maybe a link is better than no link. I'm not sure

Reply:
Same as above, I removed the link to avoid confusion.